frontend/src/edit.jsx

```diff
@@ -1,21 +1,142 @@
-// import Dropdown from 'react-bootstrap/Dropdown';
+import React, { useState, useEffect } from 'react';
+
+import {
+  Radio,
+  RadioGroup,
+  Tab,
+  Tabs,
+} from '@blueprintjs/core';
+
+import {
+  doWallpaper,
+  makeScenario,
+  getLulcTableForParcel,
+  getWallpaperResults,
+  getStatus,
+} from './requests';
+import useInterval from './hooks/useInterval';
+import ScenarioTable from './scenarioTable';
+import ParcelTable from './parcelTable';
 
 export default function EditMenu(props) {
+  const {
+    open,
+    parcelCoords,
+    parcelID,
+    savedScenarios,
+    refreshSavedScenarios,
+  } = props;
 
+  const [activeTab, setActiveTab] = useState('create');
+  const [scenarioName, setScenarioName] = useState('');
+  const [scenarioID, setScenarioID] = useState(null);
+  const [pattern, setPattern] = useState('');
+  const [parcelTable, setParcelTable] = useState(null);
+  const [jobID, setJobID] = useState(null);
 
-  const { open } = props
-  console.log('render edit menu', open);
+  useEffect(async () => {
+    if (parcelID) {
+      const table = await getLulcTableForParcel(parcelCoords);
+      setParcelTable(table);
+    }
+  }, [parcelID]);
 
-  if (open) {
-    return (
-      <div className={"menu-container"}>
-        <span>Wallpaper with pre-made></span>
-        {/*<Dropdown>
+  useInterval(async () => {
+    console.log('checking status for', jobID);
+    const status = await getStatus(jobID);
+    if (status === 'complete') {
+      const results = await getWallpaperResults(jobID);
+      setParcelTable(results);
+      setJobID(null);
+    }
+  }, jobID ? 1000 : null);
 
-        </Dropdown>*/}
-      </div>
-    );
-  } else {
-    return <div />
+  async function handleSubmitNew(event) {
+    event.preventDefault();
+    if (!scenarioName) {
+      alert('no scenario was selected');
+      return;
+    }
+    if (!pattern) {
+      alert('no modification options selected; no changes to make');
+      return;
+    }
+    if (!parcelID) {
+      alert('no parcel was selected; no changes to make');
+      return;
+    }
+    let sid = scenarioID;
+    if (!Object.values(savedScenarios).includes(scenarioName)) {
+      sid = await makeScenario(scenarioName, 'description');
+      setScenarioID(sid);
+    }
```

Comment:
Good question. ID is unique, as that's created by the backend database. The way it's designed now, name is also unique within a user's list of saved scenarios. 

So far this is missing from the app, but `savedScenarios` is always specific to the user. So eventually we'll have some initial step when the app loads where the user sends a token and the client receives `savedScenarios` for that user.

Reply:
@emlys I feel like there might be a followup question here that I'm missing?

Reply:
Just making sure since we're relying on both name and id being unique here. I think this is fine - so a scenario ID would be unique among all scenarios, and a scenario name would be unique only among one user's scenarios. If we wanted to consolidate them, maybe the id could be `<user id>.<scenario name>`